libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { useLocation, useParams, useSearchParams } from 'react-router-dom'
+import { useFetchEnvironments } from '@qovery/domains/environment'
+import { EventQueryParams, useFetchEventTargets } from '@qovery/domains/event'
+import { selectProjectsEntitiesByOrgId } from '@qovery/domains/projects'
+import { convertDatetoTimestamp } from '@qovery/shared/utils'
+import { RootState } from '@qovery/store'
+import CustomFilter from '../../ui/custom-filter/custom-filter'
+import { extractEventQueryParams, hasEnvironment } from '../page-general-feature/page-general-feature'
+
+export interface CustomFilterFeatureProps {
+  handleClearFilter: () => void
+}
+
+export function CustomFilterFeature({ handleClearFilter }: CustomFilterFeatureProps) {
+  const location = useLocation()
+  const { organizationId = '' } = useParams()
+  const [, setSearchParams] = useSearchParams()
+  const [timestamps, setTimestamps] = useState<[Date, Date] | undefined>()
+  const [isOpenTimestamp, setIsOpenTimestamp] = useState(false)
+
+  const queryParams: EventQueryParams = extractEventQueryParams(location.pathname + location.search)
+  const { environmentId, projectId, targetType, targetId } = queryParams
+
+  const projects = useSelector((state: RootState) => selectProjectsEntitiesByOrgId(state, organizationId))
+  const { data: environments } = useFetchEnvironments(projectId || '')
+
+  const displayEventTargets: boolean = Boolean(
+    targetType && (!hasEnvironment(targetType) || (projectId && environmentId))
+  )
+
+  const { data: eventsTargetsData, isLoading: isLoadingEventsTargetsData } = useFetchEventTargets(
+    organizationId,
+    queryParams,
+    displayEventTargets
+  )
+
+  useEffect(() => {
+    const newQueryParams: EventQueryParams = extractEventQueryParams(location.pathname + location.search)
+
+    if (newQueryParams.fromTimestamp && newQueryParams.toTimestamp)
+      setTimestamps([
+        new Date(parseInt(newQueryParams.fromTimestamp, 10) * 1000),
```

Comment:
Not very efficient queryParams management.
[should be tackled in technical task]